iothub_client/inc/iothub_device_client_ll.h

```diff
@@ -374,6 +374,24 @@ typedef struct IOTHUB_CLIENT_CORE_LL_HANDLE_DATA_TAG* IOTHUB_DEVICE_CLIENT_LL_HA
 
 #endif /*DONT_USE_UPLOADTOBLOB*/
 
+    /**
+    * @brief    This API sends an acknowledgement to Azure IoT Hub that a cloud-to-device message has been received.
+    *
+    * @param    device_ll_handle                The handle created by a call to a create function.
+    * @param    message                         The cloud-to-device message received through the callback provided to IoTHubDeviceClient_LL_SetMessageCallback.
+    * @param    disposition                     Acknowledgement option for the message.
+    *
+    * @warning  This function is to be used only when IOTHUBMESSAGE_ASYNC_ACK is used in the callback for incoming Cloud-to-Device messages.
+    * @remarks
+    *           For Cloud-to-Device messages, the IOTHUB_MESSAGE_HANDLE instance is allocated by the Azure IoT C SDK
```

Comment:
OK so I *love* this note and we're thinking about memory lifetimes and helping customer understand.  I just worry that the who own what isn't entirely clear, along same lines as Dane feedback of "it" unclear.  

I think we should maybe make it more explicit what we want the app to do (namely call this function) so we lead with headline and not get into the details as much.  (We're open source so if they're curious on this level of detail they can read themselves:)).

Maybe something like
* If your cloud-to-device message callback returned IOTHUBMESSAGE_ASYNC_ACK, it MUST call this API eventually.
*  Beyond sending acknowledgment to the service, this method also handles freeing message's memory.
* Not calling this function will result in memory leaks.

Reply:
I ended up adding the following documentation:

```c
    * @remarks
    *           If your cloud-to-device message callback returned IOTHUBMESSAGE_ASYNC_ACK, it MUST call this API eventually.
    *           Beyond sending acknowledgment to the service, this method also handles freeing message's memory.
    *           Not calling this function will result in memory leaks.
    *           Depending on the protocol used, this API will acknowledge cloud-to-device messages differently:
    *           AMQP: A MESSAGE DISPOSITION is sent using the `disposition` option provided.
    *           MQTT: A PUBACK is sent if `disposition` is `IOTHUBMESSAGE_ACCEPTED`. Passing any other option results in no PUBACK sent for the message.
    *           HTTP: A HTTP request is sent using the `disposition` option provided.
```